staging/src/k8s.io/dynamic-resource-allocation/structured/internal/incubating/pools_incubating.go

```diff
@@ -107,13 +122,30 @@ func GatherPools(ctx context.Context, slices []*resourceapi.ResourceSlice, node
 
 	}
 
+	if len(slicesWithBindingConditions) > 0 {
+		for _, slice := range slicesWithBindingConditions {
+			if err := addSlice(pools, slice); err != nil {
+				return nil, fmt.Errorf("failed to add node slice %s: %w", slice.Name, err)
+			}
+		}
+	}
```

Comment:
Condition is redundant:
```suggestion
	for _, slice := range slicesWithBindingConditions {
		if err := addSlice(pools, slice); err != nil {
			return nil, fmt.Errorf("failed to add node slice %s: %w", slice.Name, err)
		}
	}
```